src/JeffreyGacha.ts

```diff
@@ -0,0 +1,173 @@
+        'https://imgur.com/LijfPss.png',
+        'https://imgur.com/GnnuMI5.png',
+        'https://imgur.com/eGPRhm5.png',
+        'https://imgur.com/l8uOmAq.png'        
+    ],
+    Legendary: [
+        { link: 'https://imgur.com/miHJetv.png', name: 'RadiantJeffrey' },
+        { link: 'https://imgur.com/tbS4faL.png', name: 'CannibalJeffrey' },
+        { link: 'https://imgur.com/5QfM3zQ.png', name: 'DeadJeffrey' },
+        { link: 'https://imgur.com/xhWXOIA.png', name: 'BoxJeffrey' },
+        { link: 'https://imgur.com/okvejbp.png', name: 'DisturbedJeffrey' },
+        { link: 'https://imgur.com/sdLYctZ.png', name: 'CuddleJeffrey' },
+        { link: 'https://imgur.com/CYvTDSC.png', name: 'ThatsNotJeffrey!?' },
+        { link: 'https://imgur.com/jmKtrlf.png', name: 'AtpeaceJeffrey' },
+        { link: 'https://imgur.com/Y1i2gMP.png', name: 'AttackJeffrey' },
+        { link: 'https://imgur.com/3aWCqqe.png', name: 'CaughtJeffrey' },
+        { link: 'https://imgur.com/eTGaGLM.png', name: 'LapCatJeffrey' },
+        { link: 'https://imgur.com/7YdKWjS.png', name: 'GateKeeperJeffrey' }
+    ]
+    
+}
+//Object containing the Names and Descriptions of all the legendary gachas
+const LegendaryInfo: { [key: string]: { Name: string; Description: string } } = {
+    RadiantJeffrey: {
+        Name: 'Radiant Jeffrey',
+        Description: 'Our Lord and Savior Jeffrey'
+    },
+    CannibalJeffrey: {
+        Name: 'Cannibal Jeffrey',
+        Description: 'OK FINE ILL FEED YOU NOW!'
+    },
+    DeadJeffrey: {
+        Name: 'Dead Jeffrey',
+        Description: 'RIP. Fly high king'
+    },
+    BoxJeffrey: {
+        Name: 'Box Jeffrey',
+        Description: 'Wow, soooo original Jeffrey. Never seen a cat do THAT before.'
+    },
+    DisturbedJeffrey: {
+        Name: 'Disturbed Jeffrey',
+        Description: 'I don\'t think he\'s a fan...'
+    },
+    CuddleJeffrey: {
+        Name: 'Cuddle Jeffrey',
+        Description: 'Wittle Cutie!'
+    },
+    ThatsNotJeffrey: {
+        Name: 'Not Jeffrey',
+        Description: 'Hey, that\'s not Jeffrey!'
+    },
+    AtPeaceJeffrey: {
+        Name: 'Peaceful Jeffrey',
+        Description: 'That\'s about as peacful as its gonna get...'
+    },
+    AttackJeffrey: {
+        Name: 'Attack Jeffrey',
+        Description: 'That bite\'s lethal!'
+    },
+    CaughtJeffrey: {
+        Name: 'Caught Jeffrey',
+        Description: 'Hey, what going on over here!'
+    },
+    LapCatJeffrey: {
+        Name: 'Lap-Cat Jeffrey',
+        Description: 'He\'s been know to peruse a lap or two.'
+    },
+    GatekeeperJeffrey: {
+        Name: 'Gatekeeper Jeffrey',
+        Description: 'Come on Jeffrey, that\'s not very nice!'
+    }
+}
+
+/**
+ *
+ * @param gacha - takes a legendary gacha image URL 
+ * @returns - gachaInfo, contains legendary info (Name and Description)
+ */
+export async function displayLegendary(gacha: string): Promise<string[] | null> { 
+    for(let i = 0; i < 12; i++){
```

Comment:
Don't hardcode the length

```suggestion
    for(let i = 0; i < JeffreyGachaURLs["Legendary"].length; i++){
```

You can also use a for-each loop:
```ts
for (const gacha of JeffreyGachaURLs["Legendary"]) {
    if (gacha.link === ...) { ... }
}
```